services/steam/steam.service.js

```diff
@@ -0,0 +1,395 @@
+    return this._requestJson({
+      url,
+      schema: steamCollectionSchema,
+      errorMessages: {
+        400: 'bad request',
+      },
+      options: {
+        method: 'POST',
+        form: {
+          collectioncount: '1',
+          'publishedfileids[0]': collectionId,
+        },
+      },
+    })
+  }
+
+  static render({ size }) {
+    return { message: metric(size), color: 'green' }
+  }
+
+  async handle({ collectionId }) {
+    const json = await this.fetch({ collectionId })
+    if (json.response.collectiondetails[0].result === 1) {
+      return this.constructor.render({
+        size: json.response.collectiondetails[0].children.length,
+      })
+    } else {
+      return { message: 'collection not found', color: 'red' }
+    }
+  }
+
+  static get category() {
+    return 'other'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'files' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/collection-files',
+      format: '(.+)',
+      capture: ['collectionId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam Collection Files',
+        exampleUrl: '180077636',
+        urlPattern: ':collection_id',
+        staticExample: this.render({ size: 32 }),
+        keywords: ['steam'],
+      },
+    ]
+  }
+}
+
+class SteamFileService extends BaseJsonService {
+  async fetch({ fileId }) {
+    const url =
+      'https://api.steampowered.com/ISteamRemoteStorage/GetPublishedFileDetails/v1/?format=json'
+    return this._requestJson({
+      url,
+      schema: steamFileSchema,
+      errorMessages: {
+        400: 'bad request',
+      },
+      options: {
+        method: 'POST',
+        form: {
+          itemcount: 1,
+          'publishedfileids[0]': fileId,
+        },
+      },
+    })
+  }
+
+  async handle({ fileId }) {
+    const json = await this.fetch({ fileId })
+
+    if (json.response.publishedfiledetails[0].result === 1) {
+      return this.onRequest({ response: json.response.publishedfiledetails[0] })
+    } else {
+      return { message: 'file not found', color: 'red' }
+    }
+  }
+
+  async onRequest({ response }) {}
+
+  static get defaultBadgeData() {
+    return { label: 'steam' }
+  }
+
+  static get category() {
+    return 'other'
+  }
+}
+
+class SteamFileSize extends SteamFileService {
+  static render({ fileSize }) {
+    return { message: prettyBytes(fileSize), color: 'green' }
+  }
+
+  async onRequest({ response }) {
+    return this.constructor.render({ fileSize: response.file_size })
+  }
+
+  static get category() {
+    return 'size'
+  }
+
+  static get defaultBadgeData() {
+    return { label: 'size' }
+  }
+
+  static get url() {
+    return {
+      base: 'steam/size',
+      format: '(.+)',
+      capture: ['fileId'],
+    }
+  }
+
+  static get examples() {
+    return [
+      {
+        title: 'Steam File Size',
+        exampleUrl: '100',
+        urlPattern: ':file_id',
```

Comment:
Minor style point: I think we're mostly using `camelCase` for the url params in preference to `snake_case`